### PR TITLE
Cut ABR segments more accurately when the frame duration is variable

### PR DESCRIPTION
**Pull request by elv-reza**
issue: https://github.com/qluvio/avpipe/issues/354

Testing done:
- Manual tests with etx
- Testing FOX live streams

### CODE REVIEW

File: libavformat/dashenc.c [hunk 1]
```diff
@@ -2013,11 +2017,32 @@ static int dash_write_packet(AVFormatContext *s, AVPacket *pkt)
...
+            s->max_pts_jump = delta;
+    }
+
+    int64_t frame_duration_variation = 0;
+
+    if (s->prev_pts != AV_NOPTS_VALUE)
+        frame_duration_variation = (s->max_pts_jump - s->min_pts_jump) * 2;
+    
+    if (frame_duration_variation == 0)
+        frame_duration_variation = 1;
```
Comment (elv-serban) [inline libavformat/dashenc.c:2035]: What does this do? (number '1'?)

File: libavformat/dashenc.c [hunk 2]
```diff
@@ -2013,11 +2017,32 @@ static int dash_write_packet(AVFormatContext *s, AVPacket *pkt)
...
+        if (delta < s->min_pts_jump)
+            s->min_pts_jump = delta;
+        if (delta > s->max_pts_jump)
+            s->max_pts_jump = delta;
+    }
+
+    int64_t frame_duration_variation = 0;
+
+    if (s->prev_pts != AV_NOPTS_VALUE)
+        frame_duration_variation = (s->max_pts_jump - s->min_pts_jump) * 2;
```
Comment (elv-serban) [inline libavformat/dashenc.c:2032]: Shouldn't frame variation just be max jump?

File: libavformat/dashenc.c (hunk 1 above)
Comment (elv-reza) [reply libavformat/dashenc.c:2035]: This is just for backward compatibility not to break anything unexpected (we used to have frame_duration_variation = 1).

File: libavformat/dashenc.c (hunk 2 above)
Comment (elv-reza) [reply libavformat/dashenc.c:2032]: No, imagine all the frames duration are the same and we jump 1000 pts every time. Then max_pts_jump would be equal to min_pts_jump and the difference is 0.
The other case is when the frame duration is not constant and frame pts jumps sometime 272 and sometimes 256. In this case max_pts_jump would be 272 and min_pts_jump would be 256 and the frame_duration_variant = (272-256)*2 = 32

**Review by elv-serban — APPROVED** [review]

**Pull request merged**